bot/src/yt_link_changer.py

```diff
@@ -0,0 +1,42 @@
+import re
+import typing as t
+
+
+from youtubesearchpython import VideosSearch
+from bot.config import Config
+from bot.errors import filter
+
+
+class YTLinks:  # вынести это говно  в SRC кудато
+    watch_link = r'^https://www\.youtube\.com/watch\?v='
+    shorts_link = r'^https://www\.youtube\.com/shorts/'
+    base_link = r'^https://youtu\.be.*'
+    base_str = 'https://youtu.be/'
+
+    def __new__(cls, *args, **kwargs):
+        if not hasattr(cls, '_instance'):
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def link_rebaser(self, link: str, regex: str) -> str:
+        return re.sub(regex, self.base_str, link)
+
+    def yt_search(self, text: str, res_len: int = 3) -> str:
+        videosSearch = VideosSearch(text, limit=res_len)
+        search_result = videosSearch.result()
+        video_titles = []
+        video_links = []
+        for idx, res in enumerate(search_result.get('result')):
+            video_titles.append(f"{idx + 1}. {res.get('title')} \n")
+            video_links.append(f"{res.get('link')}")
+        return video_titles, video_links
+
+    def filter(self, text: str) -> t.Optional[str]:
+        if re.match(self.watch_link, text):
+            return self.link_rebaser(text, self.watch_link)
+        elif re.match(self.shorts_link, text):
+            return self.link_rebaser(text, self.shorts_link)
+        elif re.match(self.base_link, text):
+            return text
```

Comment:
implement there `link_extractor`

Reply:
![image](https://user-images.githubusercontent.com/71280626/195418926-07f1033f-a25d-41bb-8c98-2458c215042f.png)
Unga don't definitely understand what is meaning there